R/LSgaussianestimation.R

```diff
@@ -197,135 +195,151 @@ LSgaussianestimation   <- function(jaspResults, dataset, options, state = NULL){
         dfLinesPP   <- dfLinesPP[dfLinesPP$g == type,]
         dfLinesPP$g <- options[["priors"]][[i]]$name
 
-        all_lines   <- c(all_lines, list(dfLinesPP))
+        allLines   <- c(allLines, list(dfLinesPP))
         legend      <- rbind(legend, c(options[["priors"]][[i]]$type, options[["priors"]][[i]]$name))
 
       }
     }
 
     xName  <- bquote(.(gettext("Population mean"))~mu)
 
-    if(options[[ifelse(type == "Prior", "plotsPriorType", "plotsPosteriorType")]] == "overlying"){
-      p <- .plotOverlyingLS(all_lines, all_arrows, xName = xName, palette = options[["colorPalette"]], xRange = range)
-    }else{
-      p <- .plotStackedLS(all_lines, all_arrows, legend, xName = xName, xRange = range)
+    if (options[[ifelse (type == "Prior", "plotsPriorType", "plotsPosteriorType")]] == "overlying") {
+      p <- .plotOverlyingLS(allLines, allArrows, xName = xName, palette = options[["colorPalette"]], xRange = range)
+    } else{
+      p <- .plotStackedLS(allLines, allArrows, legend, xName = xName, xRange = range)
     }
 
     plotsSimple$plotObject <- p
   }
 
   return()
 }
-.plotsIndividualGaussianLS         <- function(jaspResults, data, ready, options, type = c("Prior", "Posterior")){
+.plotsIndividualGaussianLS         <- function(jaspResults, data, ready, options, type = c("Prior", "Posterior")) {
 
-  containerPlots <- .containerPlotsLS(jaspResults, options, "gauss_est", type)
+  containerPlots <- .containerPlotsLS(jaspResults, options, "gaussEst", type)
 
-  if(is.null(containerPlots[[paste0("plots",type,"individual")]])){
+  if (is.null(containerPlots[[paste0("plots",type,"individual")]])) {
 
     plotsIndividual <- createJaspContainer()
 
     plotsIndividual$position <- 2
-    plotsIndividual$dependOn(c(.GaussianLS_data_dependencies,
-                               ifelse(type == "Prior", "plotsPriorIndividualCI", "plotsPosteriorIndividualCI"),
-                               ifelse(type == "Prior", "plotsPriorCoverage",     "plotsPosteriorCoverage"),
-                               ifelse(type == "Prior", "plotsPriorLower",        "plotsPosteriorLower"),
-                               ifelse(type == "Prior", "plotsPriorUpper",        "plotsPosteriorUpper")))
+    plotsIndividual$dependOn(c(
+      .dataDependenciesGaussianLS,
+      if (type == "Prior") {
+        c("plotsPriorIndividualEstimate", "plotsPriorIndividualEstimateType", "plotsPriorIndividualCI",
+          "plotsPriorIndividualType", "plotsPriorCoverage", "plotsPriorLower", "plotsPriorUpper")
+      } else if (type == "Posterior") {
+        c("plotsPosteriorIndividualEstimate", "plotsPosteriorIndividualEstimateType", "plotsPosteriorIndividualCI",
+          "plotsPosteriorIndividualType", "plotsPosteriorCoverage", "plotsPosteriorLower", "plotsPosteriorUpper",
+          "plotsPosteriorBF", "plotsPosteriorIndividualPrior", "plotsPosteriorIndividualProportion")
+      }
+    ))
```

Comment:
so this is an improvement compared to the last version, but I would still try to do it in sequential steps rather than mix this all together in one spaghetti statement.

What I mean is to do the logic *outside* of the `$dependOn()`, something like this

```suggestion
    if (type == "Prior") {
        dependencies <- c("plotsPriorIndividualEstimate", "plotsPriorIndividualEstimateType", "plotsPriorIndividualCI",
          "plotsPriorIndividualType", "plotsPriorCoverage", "plotsPriorLower", "plotsPriorUpper")
    } else if (type == "Posterior") {
        dependencies <- c("plotsPosteriorIndividualEstimate", "plotsPosteriorIndividualEstimateType", "plotsPosteriorIndividualCI",
          "plotsPosteriorIndividualType", "plotsPosteriorCoverage", "plotsPosteriorLower", "plotsPosteriorUpper",
          "plotsPosteriorBF", "plotsPosteriorIndividualPrior", "plotsPosteriorIndividualProportion")
    }
    
    plotsIndividual$dependOn(c(.dataDependenciesGaussianLS, dependencies))
```

which should be much easier to parse when you read the code

Reply:
well, prefer my original version but I changed changed